pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -320,37 +310,12 @@ void processArguments() throws Exception {
                 functionConfig.setUserConfig(new HashMap<>());
             }
 
-            if (functionConfig.getInputs().isEmpty() && functionConfig.getCustomSerdeInputs().isEmpty()) {
-                throw new RuntimeException("No input topic(s) specified for the function");
-            }
-
-            // Ensure that topics aren't being used as both input and output
-            verifyNoTopicClash(functionConfig.getInputs(), functionConfig.getOutput());
-
-            if (parallelism == null) {
-                if (functionConfig.getParallelism() == 0) {
-                    functionConfig.setParallelism(1);
-                }
-            } else {
-                int num = Integer.parseInt(parallelism);
-                if (num <= 0) {
-                    throw new IllegalArgumentException("The parallelism factor (the number of instances) for the function must be positive");
-                }
-                functionConfig.setParallelism(num);
+            if (parallelism != null) {
```

Comment:
dont we need to check if parallelism < 0

Reply:
This all done via annotations.  Please seeing FunctionConfig